src/bindings/c/src/ov_compiled_model.cpp

```diff
@@ -173,27 +175,20 @@ ov_status_e ov_compiled_model_create_infer_request(const ov_compiled_model_t* co
     return ov_status_e::OK;
 }
 
-inline ov_status_e ov_compiled_model_properies_to_anymap(const ov_properties_t* properties, ov::AnyMap& dest) {
-    if (!properties || properties->size <= 0) {
+ov_status_e ov_compiled_model_set_property(const ov_compiled_model_t* compiled_model, ...) {
+    if (!compiled_model) {
         return ov_status_e::INVALID_C_PARAM;
     }
 
-    return ov_status_e::NOT_IMPLEMENTED;
-}
+    try {
+        ov::AnyMap property = {};
 
-ov_status_e ov_compiled_model_set_property(const ov_compiled_model_t* compiled_model, const ov_properties_t* property) {
-    if (!compiled_model || !property) {
-        return ov_status_e::INVALID_C_PARAM;
-    }
+        va_list args_ptr;
+        va_start(args_ptr, compiled_model);
+        GET_PROPERTY_FROM_ARGS_LIST;
```

Comment:
I think it will crash if `va_list.size() % 2 != 0` and we need to return an error status code in this case

Reply:
sure, consider this exception I think it is necessary to add the additional argument to pass the size of arg lists.

Reply:
Actually, we don't have any good method to runtime check/get the actual argument number.

Reply:
I have done some tests, it seem it will not report any error except it try to read some registers (such as pc).
The va_args() mechanism is not strict to access the args list.

Reply:
Can it introduce potential crashes or security issues?

Reply:
If caller pass invalid/incorrect parameters, it does bring some crashed or security issues, for example:

Correct calling:
`         ov_compiled_model_set_property(compiled_model, const char *key, const char *value);`

Incorrect calling:
```
         ov_compiled_model_set_property(compiled_model,  const char *key);
         ov_compiled_model_set_property(compiled_model,  const char *key, double value);
```
 
The original cause comes from va_arg() macro doesn’t understand if it reaches the end of the optional argument list.
So we need callers follow some convention to avoid above issues.

Reply:
Ok got it. 
I think in this case well documented functions should be enough.